wandb/internal/file_stream.py

```diff
@@ -73,19 +75,25 @@ class CRDedupeFilePolicy(DefaultFilePolicy):
     """
 
     def process_chunks(self, chunks):
-        content = []
-        for line in [c.data for c in chunks]:
-            if content and content[-1].endswith('\r'):
-                content[-1] = line
-            else:
-                content.append(line)
+        ret = []
+        flag = False
+        for c in chunks:
+            lines = c.data.split(os.linesep)
+            for line in lines:
+                line = line.split('\r')[-1]
+                if line:
+                    if line.endswith('\x1b\x5b\x41'):
+                        if flag:
+                            ret.pop()
+                            flag = False
+                    else:
+                        ret.append(line + os.linesep)
+                        flag = True
         chunk_id = self._chunk_id
-        self._chunk_id += len(content)
-        if content and content[-1].endswith('\r'):
-            self._chunk_id -= 1
+        self._chunk_id += len(ret)
```

Comment:
This seems wrong.  There are other libraries besides tqdm that just keep writing the same line with a carriage return to indicate progress.  That why we used to just overwrite the previous chunk in this case.  With this new logic I think we'll stream all the lines.  I would try something like the following:

```
spinner_states = ['-', '\\', '|', '/']
for s in spinner_states:
    sys.stdout.write(s)
    sys.stdout.flush()
print("")
```

And ensure that only sends one line to wandb.

Reply:
Only 1 line gets sent:

```python
spinner_states =['-', '\\', '|', '/']
for s in spinner_states * 10:
    sys.stdout.write('\r')
    sys.stdout.write(s)
    sys.stdout.flush()
    time.sleep(0.05)
print("")

```

Reply:
👍